core/services/pipeline/runner_test.go

```diff
@@ -498,7 +498,8 @@ ds1 -> ds1_parse;
 		require.NoError(t, err)
 		assert.Equal(t, jb.MaxTaskDuration, models.Interval(cltest.MustParseDuration(t, "1s")))
 
-		config.Config.Set("P2P_LISTEN_PORT", 2000) // Required to create job spawner delegate.
+		config.Config.Set("P2P_LISTEN_PORT", 2000)           // Required to create job spawner delegate.
+		config.Config.Set("P2P_PEER_ID", ek.PeerID.String()) // Required to create job spawner delegate.
```

Comment:
_I think_ this shouldn't be needed

Reply:
It is needed because we only autoset the config if we start the whole application - and we don't do that here.